### PR TITLE
[camerax] Mock `InstanceManager` in unit tests

### PR DESCRIPTION
**Pull request by camsim99**
Ensures that `InstanceManager` is mocked (once) for all `camera_android_camerax` tests.

Fixes https://github.com/flutter/flutter/issues/127661.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera_android_camerax/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Updates tests to mock `InstanceManager` to fix unit test flake.
```
Comment (stuartmorgan) [inline packages/camera/camera_android_camerax/CHANGELOG.md:3]: This doesn't need a changelog change, since it's dev-only.

**Review by stuartmorgan — COMMENTED** [review]
Can you elaborate a bit on what the issue is and how this fixes it? I'm not sure what exactly the effect of this PR is, and usually test setup being global rather than per-test *causes* issues

**Comment by camsim99** [comment]
> Can you elaborate a bit on what the issue is and how this fixes it? I'm not sure what exactly the effect of this PR is, and usually test setup being global rather than per-test _causes_ issues

I think calls to an `InstanceManager`'s method `clear()` is causing the flake, and by mocking for all tests, we ensure that it doesn't actually get called. An `InstanceManager` instance is used in essentially all the tests except for those testing detached constructors, so I think it makes sense to mock it globally, and shouldn't cause any issues.

I would defer to @bparrishMines, though. Any feedback on this?

**Review by bparrishMines — APPROVED** [review]
LGTM

**Pull request merged**